src/store/modules/Configuration/NetworkSettingsStore.js

```diff
@@ -0,0 +1,109 @@
+import api from '../../api';
+
+const NetworkSettingsStore = {
+  namespaced: true,
+  state: {
+    networkData: null,
+    ipAddress: '--',
+    macAddress: '--'
+  },
+  getters: {
+    networkData(state) {
+      return state.networkData;
+    },
+    ipAddress(state) {
+      return state.ipAddress;
+    },
+    macAddress(state) {
+      return state.macAddress;
+    }
+  },
+  mutations: {
+    setNetworkData(state, networkData) {
+      state.networkData = networkData;
+    },
+    setIpAddress(state, ipAddress) {
+      state.ipAddress = ipAddress;
+    },
+    setMacAddress(state, macAddress) {
+      state.macAddress = macAddress;
+    }
+  },
+  actions: {
+    getNetworkData({ commit }) {
+      api
+        .get('/xyz/openbmc_project/network/enumerate')
+        .then(({ data }) => {
+          const networkData = JSON.stringify(data.data);
+          const networkDataContent = JSON.parse(networkData);
+          let parsedNetworkData = {
+            interface_ids: [],
+            interfaces: {},
+            ip_addresses: { ipv4: [], ipv6: [] }
+          };
+          let interfaceId = '',
+            keyParts = [],
+            interfaceHash = '',
+            interfaceType = '';
+          for (let key in networkDataContent) {
+            if (key.match(/network\/eth\d+(_\d+)?$/gi)) {
```

Comment:
When we upstream, probably need some comments here on why we are doing this / what we are doing